src/lib/transaction_snark_scan_state/transaction_snark_scan_state.ml

```diff
@@ -1057,18 +1058,18 @@ let apply_ordered_txns_stepwise ?(stop_at_first_pass = false) ordered_txns
     match ordered_txns with
     | [] ->
         apply_previous_incomplete_txns previous_incomplete ~k:(fun () ->
-            Ok (`Complete (Ledger.merkle_root ledger)) )
+            Ok (`Complete first_pass_ledger_hash) )
```

Comment:
Nit: the contract of this code wasn't clear to me while reading it. It would have helped me to have the return value be of type `[First_pass_ledger_hash of Ledger_hash.t]`, which would have made this bug far more obvious.